step5

Binseop-Ko opened the pull request:
### 概要
 - Rails研修用 DB Scheme

### 実装したこと
 - Rails研修のStep5の「データ構造を考えたら、それをモデル図に手書きで書いてみましょう」
 - Rails研修のStep5の「README.md にテーブルスキーマを記載しましょう（モデル名・カラム名・データ型)」

### テスト項目・エビデンス（場合によって）
 - N/A

### 確認方法
 - N/A

BinseopKo committed:
step6

MioArai commented on myapp/README.md:
```diff
@@ -1,24 +1,107 @@
 
-* How to run the test suite
+  | Column      | Type         | Not Null? | PK | default        | description |
+-------------|-------------|--------------|--- | --- |----------------|-------------|
```
[NITS]ここでテーブルが壊れてますね

MioArai commented on myapp/README.md:
```diff
@@ -1,24 +1,107 @@
+
+* index: [board_id]
+
+### tag_tasks
```
多 vs 多の場合は複数形_複数形というルールのようです。
[[2.6 has_and_belongs_to_many関連付け](https://railsguides.jp/association_basics.html#has-and-belongs-to-many%E9%96%A2%E9%80%A3%E4%BB%98%E3%81%91)](https://railsguides.jp/association_basics.html#has-and-belongs-to-many%E9%96%A2%E9%80%A3%E4%BB%98%E3%81%91)

MioArai commented on myapp/README.md:
```diff
@@ -1,24 +1,107 @@
+* index: [board_id, status_id]
 
-* Deployment instructions
+### password_reset_keys
```
パスワードリセット機能は課題にはないのでなくてもいいかもしれません。

MioArai reviewed (commented):
ラクマはPRコメントにガッツリ書く文化のようです。
他の人のPR参考に以下のようなこと書くと良さそうです。
・概要
・実装したこと
・テスト項目・エビデンス（場合によって）
・確認方法

このステップ以降はMEチームの人を一人選んでレビューしてもらう形でいきましょう。
以下を#team_meに投稿すれば誰か一人選んでくれます。
```
friltan 公平に1回選んで kissy hayashi yuta kimura ikeda nakamura arai
```

BinseopKo committed:
[RB]

change table name
fix table

BinseopKo committed:
update erd

yuta-d-saito reviewed (commented):
[design/erd.png](https://github.com/Fablic/training/pull/1238/files#diff-147d4b4c5d609e0f02ccaf05fac4eceff76c3cbd80380fcf77e497fb3e5d2836)
この画像が何を意味しているのかわからなかったです。
もしかしてER図のまちがいでしょうか。。

テーブル構造に関しては特に現時点で言えることはないので問題ないと思います。

上記画像について、ご確認お願いします！
ご確認いただけたらApproveさせていただきます。

BinseopKo committed:
update erd

Binseop-Ko commented:
> [design/erd.png](https://github.com/Fablic/training/pull/1238/files#diff-147d4b4c5d609e0f02ccaf05fac4eceff76c3cbd80380fcf77e497fb3e5d2836) この画像が何を意味しているのかわからなかったです。 もしかしてER図のまちがいでしょうか。。
> 
> テーブル構造に関しては特に現時点で言えることはないので問題ないと思います。
> 
> 上記画像について、ご確認お願いします！ ご確認いただけたらApproveさせていただきます。

@yuta-d-saito さん。申し訳ございません。間違ったFileがcommitされたようです。erd 更新しましたので、確認お願いします。

yuta-d-saito commented:
画像が更新されていることを確認しました！
ありがとうごうございます。
ER図が見れたので、改めて確認したのですが、以下が気になりました。


1. ER図とTable definitionsが結構違うので、ER図は更新しておいた方が、後々のPRもスムーズかと思います。
2. boardの役割がわからなかったです。テーブルの役割の説明がもう少し欲しいなと思いました。 tasks,priorities,statuses,tagsにboard_idが入っているので理解が難しかったです。
4. prioritiesのtitleとsortにPKがついているのが気になりました、間違いなさそうでしょうか？
5. status_steps, tags_tasksにidがない。本来自動でついてくるので、意図的になくしているのであればその理由も知りたいです

BinseopKo committed:
[RB]

update erd
add table description for boards
add column description for board_id
fix typo (PK on title, sort @priorities)

Binseop-Ko commented:
> 画像が更新されていることを確認しました！ ありがとうごうございます。 ER図が見れたので、改めて確認したのですが、以下が気になりました。
> 
> 1. ER図とTable definitionsが結構違うので、ER図は更新しておいた方が、後々のPRもスムーズかと思います。
> 2. boardの役割がわからなかったです。テーブルの役割の説明がもう少し欲しいなと思いました。 tasks,priorities,statuses,tagsにboard_idが入っているので理解が難しかったです。
> 3. prioritiesのtitleとsortにPKがついているのが気になりました、間違いなさそうでしょうか？
> 4. status_steps, tags_tasksにidがない。本来自動でついてくるので、意図的になくしているのであればその理由も知りたいです

@yuta-d-saito さん、
ありがとうございます。修正しました。再度確認お願いします。

yuta-d-saito commented:
LGTM!
